components/emoji_picker/emoji_picker.jsx

```diff
@@ -538,6 +539,17 @@ export default class EmojiPicker extends React.PureComponent {
         ];
     }
 
+    convertEmojiToUserSkinTone = (emoji, emojiSkin, userSkinTone) => {
+        if (emojiSkin === userSkinTone) {
+            return emoji;
+        }
+
+        const newEmojiId = userSkinTone === 'default' ? emoji.unified.replace(`-${emojiSkin}`, '') : emoji.unified.replace(emojiSkin, userSkinTone);
```

Comment:
for the default sking, you can get it directly from the `source_index` property we created in `getSkinVariations` function.

When I added the code for skintones, I always thought it was going to be a one switch process, and not having to move across skintones, but given this I would suggest to add a mapping for skin variations in https://github.com/mattermost/mattermost-webapp/blob/4789a05e90fd1c000edcbca91fe2ef25695c6b50/build/emoji/emoji_gen.mjs#L165 in that loop we get the default and the skin variations, so maybe we could generate the mapping once and attach it to the emoji so we could get the variations without trying to calculate something like `emoji.variation['medium_light']` and that would return the index of the variant.

I'm not sure if this change is too big for the PR, though, 0/5

Reply:
Hey lets to that in seperate PR since we already have a ticket open for improving `emoji.mjs` file